kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3TableFactory.java

```diff
@@ -14,38 +14,76 @@
 
 import dev.responsive.kafka.api.config.ResponsiveConfig;
 import dev.responsive.kafka.internal.db.RemoteKVTable;
+import dev.responsive.kafka.internal.db.rs3.client.CreateStoreOptions;
+import dev.responsive.kafka.internal.db.rs3.client.CreateStoreOptions.ClockType;
 import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
 import dev.responsive.kafka.internal.db.rs3.client.grpc.GrpcRS3Client;
 import dev.responsive.kafka.internal.metrics.ResponsiveMetrics;
+import dev.responsive.kafka.internal.stores.TtlResolver;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.function.Supplier;
 import org.apache.kafka.common.config.ConfigException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class RS3TableFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(RS3TableFactory.class);
   private final GrpcRS3Client.Connector connector;
 
+  // kafka store names to track which stores we've created in RS3
+  private final Set<String> createdStores = new ConcurrentSkipListSet<>();
+
   public RS3TableFactory(GrpcRS3Client.Connector connector) {
     this.connector = connector;
   }
 
   public RemoteKVTable<WalEntry> kvTable(
-      final String name,
+      final String storeName,
       final ResponsiveConfig config,
+      final Optional<TtlResolver<?, ?>> ttlResolver,
       final ResponsiveMetrics responsiveMetrics,
-      final ResponsiveMetrics.MetricScopeBuilder scopeBuilder
+      final ResponsiveMetrics.MetricScopeBuilder scopeBuilder,
+      final Supplier<Integer> computeNumKafkaPartitions
   ) {
-    Map<String, String> storeIdMapping = config.getMap(
+    final Map<String, String> storeIdMapping = config.getMap(
         ResponsiveConfig.RS3_LOGICAL_STORE_MAPPING_CONFIG);
-    final String storeIdHex = storeIdMapping.get(name);
+    final String storeIdHex = storeIdMapping.get(storeName);
```

Comment:
once we switch to using a non-uuid store id, we should be able to drop this mapping and just create the store using its name right?

Reply:
yep i'll do that in a followup soon